app/controllers/api/cloud_networks_controller.rb

```diff
@@ -9,5 +9,23 @@ def delete_resource(type, resource_id, _data = {})
       task_id = cloud_network.delete_cloud_network_queue(User.current_user.userid)
       action_result(true, "Deleting #{cloud_network.name}", :task_id => task_id)
     end
+
+    def options
+      return super unless params.keys.include_any?(%w[id ems_id])
+
+      if params.key?("ems_id")
+        ems = resource_search(params["ems_id"], :ext_management_systems, ExtManagementSystem)
```

Comment:
I can never remember, but does `resource_search` handle the RBAC?

Reply:
Yes

Reply:
If the user cannot see it, do we get a nil back or does it instantly 404?

Reply:
That I don't know